javascript/utils.js

```diff
@@ -114,6 +114,29 @@ function handleErrors (response) {
   return response
 }
 
+// A proxy method to wrap a fetch call in error handling
+//
+// * url - the URL to fetch
+// * additionalHeaders - an object of additional headers passed to fetch
+//
+async function graciouslyFetch (url, additionalHeaders) {
+  try {
+    const response = await fetch(url, {
+      headers: {
+        'X-REQUESTED-WITH': 'XmlHttpRequest',
```

Comment:
Have we reached the `Mozilla/5.0` era of fetch request headers?

Reply:
@ParamagicDev recommended it to be able to do `request.xhr?` in Rails